force-app/main/default/classes/DMLOperations.cls

```diff
@@ -182,27 +274,42 @@ public with sharing class DMLOperations {
 	 */
 	public static void upsertAccountsWithContacts(List<Contact> contactsToUpsert) {
 		// Loop through each Contact
-
-		// Extract the Account Name from Contact's LastName
-
-		// Get the account based on the name or create one if it doesn't exist
-
-		// Link the Contact to its Account using the Account Id
-
-		// Upsert the Contacts
-	}
+        Map<String, Account> accountNameMap = new Map<String, Account>();   
+
+        if(!contactsToUpsert.IsEmpty()){
+            
+            for(Contact eachContact : contactsToUpsert){
+                String accountName = eachContact.LastName;// Extract the Account Name from Contact's LastName
+                        Account upsertedAccount = DMLOperations.upsertAccount(accountName);// Get the account based on the name or create one if it doesn't exist
+                        accountNameMap.put(accountName, upsertedAccount);
```

Comment:
I'm not sure I see any reason for the Map but I might be missing something. What's the reason for using this map? If it's not used for anything, you can probably remove it.